src/main/java/com/dih/connector/test/service/ConsumerOfferService.java

```diff
@@ -73,8 +76,14 @@ public void consumeOffer(UUID offerId) throws IOException {
         var permissionJsonNode = body.get("https://w3id.org/idsa/core/contractOffer").get("https://w3id.org/idsa/core/permission");
         var artifactNode = body.get("https://w3id.org/idsa/core/representation").get("https://w3id.org/idsa/core/instance");
         var agreementResponse = negotiateContract(permissionJsonNode, artifactNode, offerId);
-        var movedData = getConsumerData(agreementResponse);
-        log.info("Data: {}", movedData );
+        if (isText) {
+            var movedData = getConsumerData(agreementResponse, dataUrl -> restTemplateUtf16BEString.getForObject(dataUrl, String.class));
+            log.info("Data: {}", movedData );
+        } else {
+            byte[] data = getConsumerData(agreementResponse, dataUrl -> restTemplateDefault.getForObject(dataUrl, byte[].class));
+            String md5 = org.apache.commons.codec.digest.DigestUtils.md5Hex(data);
```

Comment:
We could import org.apache.commons.codec.digest.DigestUtils :)

Reply:
agree